src/lsst/cmservice/handlers/scripts.py

```diff
@@ -240,6 +321,23 @@ async def _write_script(
         await script.update_values(session, script_url=script_url, status=StatusEnum.prepared)
         return StatusEnum.prepared
 
+    async def _purge_products(
```

Comment:
Unusual level of repetition for your usual coding style (mr. meta!)

Was there something that made `_purge_products` difficult to push up to e.g. `ScriptHandler`?

Reply:
Stuff has different products?

Reply:
Or, prehaps, scripts have no guaranteed products

Reply:
A lot of them might have similar products.      Maybe all the real scripts at least have the script and the log file.